includes/Database_Upgrader.php

```diff
@@ -274,6 +275,59 @@ protected function rewrite_flush() {
 		}
 	}
 
+	/**
+	 * Re-index stories in Yoast SEO if permalinks are outdated.
+	 *
+	 * @return void
+	 */
+	protected function yoast_reindex_stories() {
+		if (
+			! function_exists( 'YoastSEO' ) ||
+			! class_exists( '\Yoast\WP\SEO\Repositories\Indexable_Repository' ) ||
+			! class_exists( '\Yoast\WP\SEO\Builders\Indexable_Builder' ) ||
+			! is_a( YoastSEO(), '\Yoast\WP\SEO\Main' )
+		) {
+			return;
+		}
+
+		/**
+		 * Indexable Repository.
+		 *
+		 * @var \Yoast\WP\SEO\Repositories\Indexable_Repository $repository Indexable Repository.
+		 */
+		$repository = YoastSEO()->classes->get( 'Yoast\WP\SEO\Repositories\Indexable_Repository' );
+
+		/**
+		 * Indexable Builder.
+		 *
+		 * @var \Yoast\WP\SEO\Builders\Indexable_Builder $builder Indexable Builder.
+		 */
+		$builder = YoastSEO()->classes->get( 'Yoast\WP\SEO\Builders\Indexable_Builder' );
+
+		$indexable_before = $repository->find_for_post_type_archive( Story_Post_Type::POST_TYPE_SLUG, false );
+
+		if ( ! $indexable_before || false === strpos( $indexable_before->permalink, '/web-stories//' ) ) {
+			$builder->build_for_post_type_archive( Story_Post_Type::POST_TYPE_SLUG, $indexable_before );
+		}
+
+		$all_stories = get_posts(
+			[
+				'fields'           => 'ids',
+				'suppress_filters' => false,
+				'post_type'        => [ Story_Post_Type::POST_TYPE_SLUG ],
+				'posts_per_page'   => - 1,
```

Comment:
```suggestion
				'update_post_meta_cache' => false,
				'update_post_term_cache'  => false,
				'posts_per_page'   => - 1,
```

No need to load meta caches here.

Reply:
FWIW I copied this from `includes/namespace.php`. Do we need to change that there as well?

Reply:
There is a different here, this all you want to do is get a list of IDs for posts to update them. In that other place, you want to warm caches into memory before deleting.